packages/flutter/lib/src/widgets/platform_view.dart

```diff
@@ -0,0 +1,114 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+
+import 'framework.dart';
+
+/// Embeds an Android view in the Widget hierarchy.
+///
+/// Embedding Android views is an expensive operation and should be avoided when a Flutter
+/// equivalent is possible.
+///
+/// The Android view object is created using a [PlatformViewFactory](/javadoc/io/flutter/plugin/platform/PlatformViewFactory.html).
+/// Plugins can register platform view factories with [PlatformViewRegistry#registerViewFactory](/javadoc/io/flutter/plugin/platform/PlatformViewRegistry.html#registerViewFactory-java.lang.String-io.flutter.plugin.platform.PlatformViewFactory-).
+/// Registration is typically done in the plugin's registerWith method, e.g:
+///
+/// ```java
+///   public static void registerWith(Registrar registrar) {
+///     registrar.platformViewRegistry().registerViewFactory("webview", new WebViewFactory(registrar.messenger()));
+///   }
+/// ```
+///
+/// The Android view lifetime is the same as the lifetime of the [State] object for this widget.
+/// When the [State] is disposed the platform view (and auxiliary resources) are lazily
+/// released (some resources are immediately released and some by platform garbage collector).
+class AndroidView extends StatefulWidget {
+  /// Creates a widget that embeds an Android view.
+  ///
+  /// The `viewType` parameter must not be null.
+  const AndroidView({
+    Key key,
+    @required this.viewType,
+    this.onPlatformViewCreated
+  }) : assert(viewType != null),
+       super(key: key);
+
+  /// The unique identifier for Android view type to be embedded by this widget.
+  /// A [PlatformViewFactory](/javadoc/io/flutter/plugin/platform/PlatformViewFactory.html)
+  /// for this type must have been registered.
+  ///
+  /// See also: [AndroidView] for an example of registering a platform view factory.
+  final String viewType;
+
+  /// Callback to invoke when after the Android view has been created.
+  ///
+  /// May be null.
+  final OnPlatformViewCreated onPlatformViewCreated;
+
+  @override
+  State createState() => new _AndroidViewState();
+}
+
+class _AndroidViewState extends State<AndroidView> {
+  int id;
```

Comment:
_createNewAndroidView is private, maybe these state vars should be private as well.

Reply:
done